### ISSUE TITLE
Implement the `compute_output_shape` method on MultiHeadAttention

### ISSUE THREAD

**Issue by AryaAftab** [keras-team/keras#16852]
TensorFlow version : 2.9.1
Exception encountered when calling layer TimeDistributed on model contained MultiHeadAttention.

error:
Please run in eager mode or implement the `compute_output_shape` method on your layer (MultiHeadAttention).

**Comment by sushreebarsa** [comment]
@AryaAftab 
Could you please elaborate about your feature and specify the use cases for this feature. Thank you!

**Comment by AryaAftab** [comment]
The `compute_output_shape` method is a common method in Keras layers. This method basically determines the output shape of that layer. In my usage, I passed a `tf.keras.layers.MultiHeadAttention` layer to the `tf.keras.layers.TimeDistributed` wrapper and unfortunately the wrapper was not able to handle the `MultiHeadAttention` layer. To solve this problem, the `compute_output_shape` method needs to be written for this layer.
If the `output_shape` input argument in `tf.keras.layers.MultiHeadAttention` is not specified, the output shape needs to be calculated, which is done by the `compute_output_shape` method.
For example, I rewrote the `tf.keras.layers.MultiHeadAttention` class as follows for my own work, which had an input shape of [batch_size, sequence_length, number_features].
```python
class MultiHeadAttention(tf.keras.layers.MultiHeadAttention):
    def compute_output_shape(self, input_shape):
        return (input_shape[0], input_shape[1], input_shape[2])
```
And the problem was solved.
